File: M3 Intermediate DSA/src/L7_Arrays_Subarrays/Assignment6.java
package L7_Arrays_Subarrays;

public class Assignment6 {

  public long subArraySum(int[] A) {
    int N = A.length;
    long ans = 0;

    for (int index = 0; index < N; index++) {
      long contributionCount = (long) (index + 1) * (N - index);
      ans += A[index] * contributionCount;
    }

    return ans;
  }

}

File: M3_Intermediate_DSA/L5_Subarrays/assignment6.go
package main

func SubarraySum(A []int) int64 {
	var ans int64
	N := len(A)

	for index, val := range A {
		var count int64 = int64((index + 1) * (N - index))
		ans += count * int64(val)
	}

	return ans
}
